csrc/fusion_segmenter.cpp

```diff
@@ -3399,13 +3399,24 @@ void SegmentCandidateFinder::forwardInputs() {
         continue;
       }
 
-      if (expr->output(0)->uses().size() > 1) {
+      // expr is a unary op so there is a single output. Here we look at that
+      // output's further uses
+      auto output_uses = expr->output(0)->uses();
+      if (output_uses.size() == 0) {
+        // Unused outputs terminate here
+        continue;
+      }
```

Comment:
This fixes a segfault that popped up in the `Standalone*` tests in `test_tensor_factories.cpp`. These tests have scalar inputs which are used in `FullOp`s as fill values. When the extents are zero, concretization replaces those `FullOps` with others so that the output size is hardcoded. This means the original fill value (which is now irrelevant since the output is empty) has no uses. The cast to target dtype before the `full` means we have `input -> cast -> (no uses)`, which caused a segfault here. We now just check that there's at least one use to avoid a segfault.